### PR TITLE
add nimbus-test-event definition

### PR DESCRIPTION
**Pull request by jeddai**
### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by jeddai** [comment]
# Request for data collection review form

1) What questions will you answer with this data?
Is our implementation of long-term holdback experimentation valid? See [Long-term holdback validation plan](https://docs.google.com/document/d/11r85f4QPXnQLJgJnPxfYsdcRW8Xz-djbo_vuQ-_jeFc/edit) document.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?
We need to be able to experiment with variable calls to telemetry in order to validate long-term holdback support.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
None. Data collection is required.

4) Can current instrumentation answer these questions?
No.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>nimbus-test-event</td>
    <td>Technical</td>
    <td>https://mozilla-hub.atlassian.net/browse/EXP-4221</td>
  </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
This collection is documented in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/

7) How long will this data be collected?  Choose one of the following:

This is scoped to a time-limited experiment/project until date 06-01-2024.

8) What populations will you measure?
Beta and potentially release channels.

9) If this data collection is default on, what is the opt-out mechanism for users?
The standard Glean opt-out mechanism.

10) Please provide a general description of how you will analyze this data.
We will be comparing the number of events called between various cohorts of users — those enrolled in only the holdback, those enrolled in no holdback and a control branch, and those enrolled in the treatment branches.

11) Where do you intend to share the results of your analysis?
Internally to Mozilla, specifically with the Nimbus team.

12) Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection? If so:
No

@eliserichards data-review?

File: components/nimbus/metrics.yaml [hunk 1]
```diff
@@ -257,6 +257,19 @@ nimbus_events:
       - project-nimbus@mozilla.com
     expires: never
     disabled: true
+  nimbus_test_event:
```
Comment (jhugman) [inline components/nimbus/metrics.yaml:260]: Nit: naming; suggest `finished_launching`, or `is_ready`.

**Review by jhugman — APPROVED** [review]
This looks fine; I do have a question about event naming.

File: components/nimbus/metrics.yaml (hunk 1 above)
Comment (jeddai) [reply components/nimbus/metrics.yaml:260]: So this event is specifically to support validation of holdback experiments — maybe something like `holdback_validation_is_ready`?

Comment (eliserichards) [reply components/nimbus/metrics.yaml:260]: It's being used for the holdback, but the event itself is pretty generic, it seems. I like `is_ready` or something more generic. I also think the event description can be updated:
"A(n) ~test~ event sent when Nimbus finishes launching."

Comment (jeddai) [reply components/nimbus/metrics.yaml]: Updated!

**Comment by eliserichards** [comment]
> # Request for data collection review form
> 
>     1. What questions will you answer with this data?
>        Is our implementation of long-term holdback experimentation valid? See [Long-term holdback validation plan](https://docs.google.com/document/d/11r85f4QPXnQLJgJnPxfYsdcRW8Xz-djbo_vuQ-_jeFc/edit) document.
> 
>     2. Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?
>        We need to be able to experiment with variable calls to telemetry in order to validate long-term holdback support.
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
>        None. Data collection is required.
> 
>     4. Can current instrumentation answer these questions?
>        No.
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> nimbus-test-event 	Technical 	https://mozilla-hub.atlassian.net/browse/EXP-4221
> 
>     6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
>        This collection is documented in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/
> 
>     7. How long will this data be collected?  Choose one of the following:
> 
> 
> This is scoped to a time-limited experiment/project until date 06-01-2024.
> 
>     8. What populations will you measure?
>        Beta and potentially release channels.
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
>        The standard Glean opt-out mechanism.
> 
>     10. Please provide a general description of how you will analyze this data.
>         We will be comparing the number of events called between various cohorts of users — those enrolled in only the holdback, those enrolled in no holdback and a control branch, and those enrolled in the treatment branches.
> 
>     11. Where do you intend to share the results of your analysis?
>         Internally to Mozilla, specifically with the Nimbus team.
> 
>     12. Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection? If so:
>         No

# Data Review Form (to be filled by Data Stewards)

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

   * Yes, in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/

2) Is there a control mechanism that allows the user to turn the data collection on and off? (Note, for data collection not needed for security purposes, Mozilla provides such a control mechanism) Provide details as to the control mechanism available.

   * Yes, through client telemetry opt-out mechanisms.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

   * Category 1: technical

5) Is the data collection request for default-on or default-off?

   * Default-on.

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

   * No

7) Is the data collection covered by the existing Firefox privacy notice? **If unsure: escalate to legal if:**

   * Yes

8) Does the data collection use a third-party collection tool? **If yes, escalate to legal.**

   * No

# Result
data-review+

**Review by eliserichards — APPROVED** [review]
data-review+

**Pull request merged**